### PR TITLE
feat(print.service) print make Pdf georeferenced

### PR DESCRIPTION
**Pull request by josee666**
**What is the current behavior?** 
https://github.com/infra-geo-ouverte/igo2/issues/747

**What is the new behavior?**
when you print a pdf is now georeference so you can put it in SIG or app like Avenza

you can find information like bbox, bound, date pdf Create and producer IGO2 on open pdf doc with blocNote.

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x ] No

### CODE REVIEW

**Review by pelord — APPROVED** [review]
validate with geomoose the licence/repo creation

**Review by pelord — CHANGES_REQUESTED** [review]
validate map projection OR project extents to 3857

File: packages/geo/src/lib/print/shared/print.service.ts [hunk 1]
```diff
@@ -115,6 +169,31 @@ export class PrintService {
...
+
+    // exemple in geoMoose
+    // let map_extents = map.viewController.olView.calculateExtent([
+    //   this.toPoints(width, unit) * resolution,
+    //   this.toPoints(height, unit) * resolution,
+    // ]);
+
+    const mapSize = map.ol.getSize();
+    const mapExtent = map.ol.getView().calculateExtent(mapSize);
+    doc.setGeoArea(pdf_extents, mapExtent);
```
Comment (pelord) [inline packages/geo/src/lib/print/shared/print.service.ts:194]: validate if mapextent are in 3857.

**Review by pelord — CHANGES_REQUESTED** [review]
validate map projection OR project extents to 3857

File: packages/geo/src/lib/print/shared/print.service.ts (hunk 1 above)
Comment (josee666) [reply packages/geo/src/lib/print/shared/print.service.ts:194]: yes ok! so if the map isn't in 3857 we do not georef ... the plugin is config for 3857.

Comment (pelord) [reply packages/geo/src/lib/print/shared/print.service.ts:194]: the georef will be written to the geopdf.. but will be wrong.

Comment (pelord) [reply packages/geo/src/lib/print/shared/print.service.ts:194]: take a look to https://openlayers.org/en/latest/apidoc/module-ol_proj.html#.transformExtent

Comment (josee666) [reply packages/geo/src/lib/print/shared/print.service.ts:194]: yes, in the plugin it work for 3857, so i will check and if isn't 3857, not reference that pdf.

Comment (josee666) [reply packages/geo/src/lib/print/shared/print.service.ts:187]: fix now getExtent is made in 3857. Thanks

**Review by PhilippeLafreniere18 — APPROVED** [review]
No more comments that @pelord already made.

For next phase, it would be interesting to fix PPI (seems to be a problem when set to higher number, quality isn't right and sometimes it shrinks the geopdf)

**Pull request merged**